api/src/main/java/org/apache/iceberg/TableScan.java

```diff
@@ -101,4 +101,13 @@ default TableScan appendsAfter(long fromSnapshotId) {
    * @return the Snapshot this scan will use
    */
   Snapshot snapshot();
+
+  /**
+   * Create a new {@link TableScan} from this scan's configuration that will have column stats
+   *
+   * @return a new scan based on this with column stats
+   */
+  default TableScan withColStats() {
```

Comment:
Why do we have to add this if we already have `includeColumnStats` defined in `Scan`?
I think we should be able to use that.

Reply:
Right. Sorry I didn't notice there is already an existing method.

Reply:
No problem at all. I forgot about that one initially too.